src/fides/api/ops/schemas/privacy_notice.py

```diff
@@ -0,0 +1,105 @@
+from __future__ import annotations
+
+from datetime import datetime
+from typing import List, Optional, Type
+
+from pydantic import conlist
+from sqlalchemy.orm import Session
+from starlette.exceptions import HTTPException
+from starlette.status import HTTP_422_UNPROCESSABLE_ENTITY
+
+from fides.api.ctl.sql_models import DataUse  # type: ignore[attr-defined]
+from fides.api.custom_types import SafeStr
+from fides.api.ops.models.privacy_notice import (
+    ConsentMechanism,
+    EnforcementLevel,
+    PrivacyNoticeRegion,
+)
+from fides.api.ops.schemas.base_class import BaseSchema
+
+
+class PrivacyNotice(BaseSchema):
+    """
+    Base for PrivacyNotice API objects
+
+    All fields are optional, since pydantic allows subclasses to be
+    stricter but not less strict
+    """
+
+    name: Optional[SafeStr]
+    description: Optional[SafeStr]
+    origin: Optional[SafeStr]
+    regions: Optional[conlist(PrivacyNoticeRegion, min_items=1)]  # type: ignore
+    consent_mechanism: Optional[ConsentMechanism]
+    data_uses: Optional[conlist(SafeStr, min_items=1)]  # type: ignore
+    enforcement_level: Optional[EnforcementLevel]
+    disabled: Optional[bool] = False
+    has_gpc_flag: Optional[bool]
+    displayed_in_privacy_center: Optional[bool]
+    displayed_in_privacy_modal: Optional[bool]
+    displayed_in_banner: Optional[bool]
+
+    class Config:
+        """Populate models with the raw value of enum fields, rather than the enum itself"""
+
+        use_enum_values = True
+        orm_mode = True
+
+    def validate_data_uses(self, valid_data_uses: List[str]) -> None:
+        """
+        Utility to validate that all specified data_uses exist as `DataUse`s.
+        Raises a ValueError if an unknown `DataUse` is found.
+        """
+        for data_use in self.data_uses or []:
+            if data_use not in valid_data_uses:
+                raise ValueError(f"Unknown data_use '{data_use}'")
+
+    @classmethod
+    def validate_notice_data_uses(
+        cls: Type[PrivacyNotice],
+        privacy_notices: List[PrivacyNotice],
+        db: Session,
+    ) -> None:
+        """
+        Ensures that all the provided `PrivacyNotice`s have valid data uses.
+        Raises a 422 HTTP exception if an unknown data use is found on any `PrivacyNotice`
+        """
+        valid_data_uses = [data_use.fides_key for data_use in DataUse.query(db).all()]
+        try:
+            for privacy_notice in privacy_notices:
+                privacy_notice.validate_data_uses(valid_data_uses)
+        except ValueError as e:
+            raise HTTPException(HTTP_422_UNPROCESSABLE_ENTITY, detail=str(e))
```

Comment:
wasn't sure if it was appropriate to throw an `HTTPException` from within the pydantic schema, but this class is pretty specifically used for API modeling so i figured it may be OK? it did make the endpoint-level code cleaner/simpler to have the `HTTPException` thrown from within this class method.